applications/OptimizationApplication/python_scripts/output/container_variable_data_holder_output_decorator.py

```diff
@@ -0,0 +1,228 @@
+            "type"                             : "ContainerVariableDataHolderOutputDecorator",
+            "output_processes"                 : [],
+            "optimization_info_output_settings": []
+        }""")
+#
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        self.__list_of_data_sets = []
+        optimization_info_output_settings: Kratos.Parameters
+        for optimization_info_output_settings in parameters["optimization_info_output_settings"]:
+            if not optimization_info_output_settings.Has("type"):
+                raise RuntimeError(f"\"type\" is not defined for optimization_info_output_settings:\n" + str(optimization_info_output_settings))
+
+            output_type = optimization_info_output_settings["type"].GetString()
+            if output_type == "response_data":
+                self.__list_of_data_sets.append(ResponseContainerVariableDataHolderInfo(optmization_info, optimization_info_output_settings))
+            elif output_type == "control_data":
+                self.__list_of_data_sets.append(ControlContainerVariableDataHolderInfo(optmization_info, optimization_info_output_settings))
+            else:
+                raise RuntimeError(f"Unsupported \"{output_type}\" found. Followings are available options:\n\tresponse_data\n\tcontrol_data")
+
+        self.__output_processes_list: 'list[Kratos.OutputProcess]' = KratosProcessFactory(model).ConstructListOfProcesses(parameters["output_processes"])
+        for output_process in self.__output_processes_list:
+            if not isinstance(output_process, Kratos.OutputProcess):
+                raise RuntimeError(f"The provided output process is not of the type Kratos.OutputProcess. Details of the process: " + str(output_process))
+
+    def ExecuteInitialize(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteInitialize)
+
+    def ExecuteBeforeSolutionLoop(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteBeforeSolutionLoop)
+
+    def ExecuteInitializeSolutionStep(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteInitializeSolutionStep)
+
+    def ExecuteBeforeOutputStep(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteBeforeOutputStep)
+
+    def ExecuteAfterOutputStep(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteAfterOutputStep)
+
+    def ExecuteFinalizeSolutionStep(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteFinalizeSolutionStep)
+
+    def ExecuteFinalize(self):
+        CallOnAll(self.__output_processes_list, Kratos.OutputProcess.ExecuteFinalize)
+
+    def IsOutputStep(self) -> bool:
+        for output_process in self.__output_processes_list:
+            if output_process.IsOutputStep():
+                return True
```

Comment:
```suggestion
        if any(output_process.IsOutputStep() for output_process in self.__output_processes_list):
            return True
```

also, this triggers the output for all processes in `self.__output_processes_list` even if only one them returns true from `IsOutputStep`. Is that intended?